ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -384,33 +384,26 @@ export function WorkflowDetails({history, location, match}: RouteComponentProps<
     // Get workflow
     useEffect(() => {
         (async () => {
-            let archivedWf: Workflow;
-            if (uid !== '') {
-                try {
-                    archivedWf = await services.workflows.getArchived(namespace, uid);
-                    setError(null);
-                } catch (err) {
-                    if (err.status !== 404) {
-                        setError(err);
-                    }
-                }
-            }
-
             try {
                 const wf = await services.workflows.get(namespace, name);
-                setError(null);
```

Comment:
`setError(null);` should still be done whenever successful -- in order to reset the error if there was one.

so both in this case and with the archived case

Reply:
How does this look? https://github.com/argoproj/argo-workflows/pull/12972/commits/37a5c8ee848d81c810f231407735f48256dfe282

Reply:
the returns aren't strictly necessary, but nbd

Reply:
Ack. I'm out of my comfort zone in TypeScript, thanks for putting up with my noobness.